mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxAnnotation.java

```diff
@@ -0,0 +1,69 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.model;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+
+/**
+ * Stores an Metadata Extension applicable to a mailbox. Inspired by
+ * RFC5464 The IMAP METADATA Extension.
+ * 
+ * Implementations must be immutable. Implementations should override
+ * {@link #hashCode()} and {@link #equals(Object)}.
+ * 
+ */
+public class MailboxAnnotation {
+
+    public static MailboxAnnotation nil(String key) {
+        return new MailboxAnnotation(key, Optional.<String>absent());
+    }
+
+    private final String key;
+    private final Optional<String> value;
+
+    public MailboxAnnotation(String key, String value) {
```

Comment:
Using Guava, you can add a precondition to ensure value is not null (and write according tests).

We usually do this with : 

```
Preconditions.checkNotNull(value, "optional message");
```